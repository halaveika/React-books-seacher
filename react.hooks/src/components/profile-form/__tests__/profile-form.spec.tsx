import React from 'react';
import Enzyme, { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import ProfileForm from '../profile-form';
Enzyme.configure({ adapter: new Adapter() });

const addCard = jest.fn();

describe('test ProfileForm component', () => {
  const wrapper = shallow(<ProfileForm addCard={addCard} />);
  jest.spyOn(window, 'alert').mockImplementation(() => {});
  it('should render ProfileForm component', async () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('form should be disabled', () => {
    const submitBtn = wrapper.find('.submit-btn');
    submitBtn.simulate('click');
    expect(window.alert).not.toHaveBeenCalled();
  });

  // it('should print firstname value', () => {
  //   const readAsBinaryStringSpy = jest.spyOn(FileReader.prototype, 'readAsBinaryString');
  //   const firstname = wrapper.find('#firstname');
  //   const lastname = wrapper.find('#lastname');
  //   const zipcode = wrapper.find('#zipcode');
  //   const birthday = wrapper.find('#birthday');
  //   const gender = wrapper.find('#gender');
  //   const avatar = wrapper.find('#avatar');
  //   const city = wrapper.find('#city');
  //   firstname.simulate('change', {target:{value: mockCardArr[0].firstname}});
  //   lastname.simulate('change', {target:{value: mockCardArr[0].lastname}});
  //   zipcode.simulate('change', {target:{value: mockCardArr[0].zipcode}});
  //   birthday.simulate('change', {target:{value: mockCardArr[0].birthday}});
  //   gender.simulate('change', {target:{value: mockCardArr[0].gender}});
  //   avatar.simulate('change', {target:{value: mockCardArr[0].avatar}});
  //   city.simulate('change', {target:{value: mockCardArr[0].city}});
  //   const submitBtn = wrapper.find('.submit-btn');
  //   submitBtn.simulate('click');
  //   expect(window.alert).toHaveBeenCalled();
  // });
});
