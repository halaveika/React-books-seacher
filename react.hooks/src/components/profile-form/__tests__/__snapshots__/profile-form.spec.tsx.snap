// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ProfileForm component should render ProfileForm component 1`] = `
<form
  className="profile-form"
  onSubmit={[Function]}
>
  <div
    className="profile-item"
  >
    <span
      className="profile-item__title"
    >
      Firstname
    </span>
    <div
      className="input-container"
    >
      <input
        id="firstname"
        name="firstname"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
      />
      <label
        className="label-validation"
        htmlFor="firstname"
      />
    </div>
  </div>
  <div
    className="profile-item"
  >
    <span
      className="profile-item__title"
    >
      Lastname
    </span>
    <div
      className="input-container"
    >
      <input
        id="lastname"
        name="lastname"
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
      />
      <label
        className="label-validation"
        htmlFor="lastname"
      />
    </div>
  </div>
  <div
    className="profile-item"
  >
    <span
      className="profile-item__title"
    >
      Zipcode
    </span>
    <div
      className="input-container"
    >
      <input
        id="zipcode"
        name="zipcode"
        onBlur={[Function]}
        onChange={[Function]}
        type="number"
      />
      <label
        className="label-validation"
        htmlFor="zipcode"
      />
    </div>
  </div>
  <div
    className="profile-item"
  >
    <span
      className="profile-item__title"
    >
      Birthday date
    </span>
    <div
      className="input-container"
    >
      <input
        id="birthday"
        name="birthday"
        onBlur={[Function]}
        onChange={[Function]}
        type="date"
      />
      <label
        className="label-validation"
        htmlFor="birthday"
      />
    </div>
  </div>
  <div
    className="profile-item"
  >
    <span
      className="profile-item__title"
    >
      Choose your city:
    </span>
    <div
      className="input-container"
    >
      <select
        id="city"
        name="city"
        onBlur={[Function]}
        onChange={[Function]}
      >
        <option
          value=""
        >
          --Please choose a city--
        </option>
        <option
          key="Minsk"
          value="Minsk"
        >
          Minsk
        </option>
        <option
          key="Brest"
          value="Brest"
        >
          Brest
        </option>
        <option
          key="Grodno"
          value="Grodno"
        >
          Grodno
        </option>
        <option
          key="Gomel"
          value="Gomel"
        >
          Gomel
        </option>
        <option
          key="Vitebsk"
          value="Vitebsk"
        >
          Vitebsk
        </option>
        <option
          key="Mogilev"
          value="Mogilev"
        >
          Mogilev
        </option>
      </select>
      <label
        className="label-validation"
        htmlFor="city"
      />
    </div>
  </div>
  <div
    className="profile-item"
  >
    <span
      className="profile-item__title"
    >
      Gender
    </span>
    <Switcher
      id="gender"
      refProp={
        Object {
          "name": "gender",
          "onBlur": [Function],
          "onChange": [Function],
          "ref": [Function],
        }
      }
    />
  </div>
  <div
    className="profile-item"
  >
    <span
      className="profile-item__title"
    >
      Avatar
    </span>
    <div
      className="input-container"
    >
      <input
        accept="image/*"
        id="avatar"
        name="avatar"
        onBlur={[Function]}
        onChange={[Function]}
        type="file"
      />
      <label
        className="label-validation"
        htmlFor="avatar"
      />
    </div>
  </div>
  <fieldset
    className="gift-checklist"
  >
    <legend>
      Choose your an additional gift
    </legend>
    <div
      key="badge"
    >
      <label
        htmlFor="badge"
      >
        badge
      </label>
      <input
        id="badge"
        name="gifts"
        onBlur={[Function]}
        onChange={[Function]}
        type="checkbox"
        value="badge"
      />
    </div>
    <div
      key="calendar"
    >
      <label
        htmlFor="calendar"
      >
        calendar
      </label>
      <input
        id="calendar"
        name="gifts"
        onBlur={[Function]}
        onChange={[Function]}
        type="checkbox"
        value="calendar"
      />
    </div>
    <div
      key="cup"
    >
      <label
        htmlFor="cup"
      >
        cup
      </label>
      <input
        id="cup"
        name="gifts"
        onBlur={[Function]}
        onChange={[Function]}
        type="checkbox"
        value="cup"
      />
    </div>
    <div
      key="pencil"
    >
      <label
        htmlFor="pencil"
      >
        pencil
      </label>
      <input
        id="pencil"
        name="gifts"
        onBlur={[Function]}
        onChange={[Function]}
        type="checkbox"
        value="pencil"
      />
    </div>
  </fieldset>
  <button
    className="submit-btn"
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;
